app/src/main/java/com/github/polybooks/database/FBSaleDatabase.kt

```diff
@@ -139,6 +139,7 @@ class FBSaleDatabase(firestore: FirebaseFirestore, private val bookDB: BookDatab
                 val booksFuture = getBookQuery().getAll()
                 return booksFuture.thenCompose { books ->  //those are the books for which we want to find the sales
                     val isbns = books.map {it.isbn}
+                    if (isbns.isEmpty()) return@thenCompose CompletableFuture.completedFuture(listOf())
```

Comment:
Can you explain how this actually works? La doc semble le faire en deux temps: 1) set un label, 2) utiliser le label. Comment est-ce que tu fais les deux en même temps?

Et comment ca fonctionne du coup? C'est vraiment une syntaxe cheloue imo. Genre y a 2 return? Le 2ème remplace le premier?